File: src/main/java/bot/inker/ankh/core/hologram/nop/NopHologramService.java
package bot.inker.ankh.core.hologram.nop;

import bot.inker.ankh.core.api.hologram.HologramContent;
import bot.inker.ankh.core.api.hologram.HologramService;
import bot.inker.ankh.core.api.hologram.HologramTask;

import javax.inject.Inject;

public class NopHologramService implements HologramService {
  @Inject
  private NopHologramService() {

  }

  @Override
  public HologramContent.Builder content() {
    return new NopHologramContent.Builder();
  }

  @Override
  public HologramTask.Builder builder() {
    return null;
  }
}

File: src/main/kotlin/bot/inker/ankh/core/hologram/nop/NopHologramService.kt
package bot.inker.ankh.core.hologram.nop

import bot.inker.ankh.core.api.hologram.HologramService
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class NopHologramService @Inject private constructor(

) : HologramService {
  override fun content() = NopHologramContent.Builder()
  override fun builder() = NopHologramTask.Builder()
}